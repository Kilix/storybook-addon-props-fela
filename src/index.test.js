import React, { PropTypes } from 'react';
import renderer from 'react-test-renderer';
import { pickProps, parseProps, renderList } from './index';

const Provider = ({ children }) => <div>{children}</div>;
const Test = ({ color, children }) => <div style={{ color }}>{children}</div>;
const TestWithDefault = ({ color, children }) => <div style={{ color }}>{children}</div>;
TestWithDefault.defaultProps = { color: '#333' };

const root = () => (
  <Provider>
    <Test color="#000">Hello</Test>
  </Provider>
);
const deepRoot = () => (
  <Provider>
    <Provider>
      <Test color="#000">Hello</Test>
    </Provider>
  </Provider>
);
const noRoot = () => <Test color="#000">Hello</Test>;
const withDefault = () => <TestWithDefault color="#000">Hello</TestWithDefault>;

test('pickProps', () => {
  const result = pickProps(TestWithDefault)(withDefault());
  const expected = {
    color: { prop: '#000', required: null, defaultProps: '#333' },
    children: { prop: 'Hello', required: null, defaultProps: undefined },
  };
  expect(result).toEqual(expected);
});

test('parseProps', () => {
  const result = parseProps({}, { cover: '#333' }, { cover: { required: true } });
  const result2 = parseProps({ cover: '#123' }, { cover: '#333' });
  expect(result).toEqual({ cover: { prop: undefined, required: true, defaultProps: '#333' } });
  expect(result2).toEqual({ cover: { prop: '#123', required: null, defaultProps: '#333' } });
});

test('renderList', () => {
  const base = {
    color: { prop: '#000', required: null, defaultProps: '#333' },
    children: { prop: 'Hello', required: null, defaultProps: undefined },
  };
  const result = renderList(base);
  const component = renderer.create(<div>{result}</div>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
