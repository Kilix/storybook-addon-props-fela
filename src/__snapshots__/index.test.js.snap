// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderList 1`] = `
<div>
  <div
    style={
      Object {
        "alignItems": "flex-start",
        "borderBottom": "1px solid rgba(175, 175, 175, .3)",
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <span
      style={
        Object {
          "flex": 1,
          "padding": "12px 5px",
        }
      }
    >
      color
    </span>
    <span
      style={
        Object {
          "flex": 1,
          "padding": "12px 5px",
        }
      }
    >
      #000
    </span>
    <span
      style={
        Object {
          "flex": 1,
          "padding": "12px 5px",
        }
      }
    >
      ✗
    </span>
    <span
      style={
        Object {
          "flex": 1,
          "padding": "12px 5px",
        }
      }
    >
      #333
    </span>
  </div>
  <div
    style={
      Object {
        "alignItems": "flex-start",
        "borderBottom": "1px solid rgba(175, 175, 175, .3)",
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <span
      style={
        Object {
          "flex": 1,
          "padding": "12px 5px",
        }
      }
    >
      children
    </span>
    <span
      style={
        Object {
          "flex": 1,
          "padding": "12px 5px",
        }
      }
    >
      Hello
    </span>
    <span
      style={
        Object {
          "flex": 1,
          "padding": "12px 5px",
        }
      }
    >
      ✗
    </span>
    <span
      style={
        Object {
          "flex": 1,
          "padding": "12px 5px",
        }
      }
    >
      -
    </span>
  </div>
</div>
`;
